### PR TITLE
netty: bump to 4.1.42

### PR DESCRIPTION
**Pull request by ppiotrow**

### CODE REVIEW

**Comment by njhill** [comment]
Probably best to go straight to 4.1.42.Final now?

**Comment by ppiotrow** [comment]
Ok, I'll update to 42. Can you share with me a simpler than `shasum -a 256` way of updating Bazel builds?

File: SECURITY.md [hunk 1]
```diff
@@ -395,7 +395,8 @@ grpc-netty version | netty-handler version | netty-tcnative-boringssl-static ver
 1.18.x-1.19.x      | 4.1.32.Final          | 2.0.20.Final
 1.20.x-1.21.x      | 4.1.34.Final          | 2.0.22.Final
 1.22.x             | 4.1.35.Final          | 2.0.25.Final
-1.23.x-            | 4.1.38.Final          | 2.0.25.Final
+1.23.x             | 4.1.38.Final          | 2.0.25.Final
+1.24.x-            | 4.1.41.Final          | 2.0.25.Final
```
Comment (ran-su) [inline SECURITY.md]: This PR did not got into the 1.24 release, therefore here still need to be 4.1.38.Final.

**Comment by creamsoup** [comment]
gradle can't resolve the new shiny 4.1.42 yet (maybe the central is behind?). i'll rerun the CI later.

**Comment by ppiotrow** [comment]
I don't know either. It is published as I was able to calculate sha256, maybe not synced to relevant mirror.
```
VERSION=4.1.42.Final
ART=netty-transport-native-epoll
wget http://central.maven.org/maven2/io/netty/$ART/$VERSION/${ART}-$VERSION.jar && shasum -a 256 ${ART}-$VERSION.jar
```

**Comment by creamsoup** [comment]
FYI. the google mirror doesn't have the 42 yet, i created a PR(#6212) to add mavenCentral as fall back to resolve this dependency.

**Comment by creamsoup** [comment]
okay now all other test passed except bazel.

```
ERROR: /tmpfs/src/github/grpc-java/netty/shaded/BUILD.bazel:2:1: //netty/shaded:shaded depends on @io_netty_netty_transport_native_epoll//jar:jar in repository @io_netty_netty_transport_native_epoll which failed to fetch. no such package '@io_netty_netty_transport_native_epoll//jar': java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-transport-native-epoll/4.1.42.Final/netty-transport-native-epoll-4.1.42.Final-linux-x86_64.jar] to /tmpfs/tmp/bazel/external/io_netty_netty_transport_native_epoll/netty-transport-native-epoll-4.1.42.Final-linux-x86_64.jar: Checksum was 7bdf3003d5b60b061b494e62d1bafc420caf800efb743b14ec01ceaef1d3fa3e but wanted 3c7d659b3bd773e0ea9b7517d2d6baffa275a3d2ae8eb4c10cb8f0a7724b11d5
ERROR: Analysis of target '//netty/shaded:shaded' failed; build aborted: no such package '@io_netty_netty_transport_native_epoll//jar': java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-transport-native-epoll/4.1.42.Final/netty-transport-native-epoll-4.1.42.Final-linux-x86_64.jar] to /tmpfs/tmp/bazel/external/io_netty_netty_transport_native_epoll/netty-transport-native-epoll-4.1.42.Final-linux-x86_64.jar: Checksum was 7bdf3003d5b60b061b494e62d1bafc420caf800efb743b14ec01ceaef1d3fa3e but wanted 3c7d659b3bd773e0ea9b7517d2d6baffa275a3d2ae8eb4c10cb8f0a7724b11d5
```

**Comment by ran-su** [comment]
I saw netty_tcnative also have a newer version, and that version got into this netty release. So it need to be upgraded in this PR too.

**Comment by ran-su** [comment]
@ppiotrow thanks for your contribution to gRPC.

**Pull request merged**